In [2]:
import json
import pandas as pd
import re

In [3]:
with open('uniprot_json.json') as uniprot_json:
    uniprot_data = json.load(uniprot_json)

In [4]:
print(uniprot_data.keys())

dict_keys(['results'])


There is only one key and that is results

In [5]:
protein_access = uniprot_data['results']

In [6]:
print(protein_access[0])

{'entryType': 'UniProtKB reviewed (Swiss-Prot)', 'primaryAccession': 'A0A075TMP8', 'uniProtkbId': 'PATI_PENEN', 'entryAudit': {'firstPublicDate': '2018-12-05', 'lastAnnotationUpdateDate': '2022-12-14', 'lastSequenceUpdateDate': '2014-10-29', 'entryVersion': 31, 'sequenceVersion': 1}, 'annotationScore': 5.0, 'organism': {'scientificName': 'Penicillium expansum', 'commonName': 'Blue mold rot fungus', 'taxonId': 27334, 'lineage': ['Eukaryota', 'Fungi', 'Dikarya', 'Ascomycota', 'Pezizomycotina', 'Eurotiomycetes', 'Eurotiomycetidae', 'Eurotiales', 'Aspergillaceae', 'Penicillium']}, 'proteinExistence': '1: Evidence at protein level', 'proteinDescription': {'recommendedName': {'fullName': {'evidences': [{'evidenceCode': 'ECO:0000303', 'source': 'PubMed', 'id': '25120234'}], 'value': 'Cytochrome P450 monooxygenase patI'}, 'ecNumbers': [{'evidences': [{'evidenceCode': 'ECO:0000269', 'source': 'PubMed', 'id': '30680886'}], 'value': '1.-.-.-'}]}, 'alternativeNames': [{'fullName': {'evidences': [{

In [7]:
protein_access[0].keys()

dict_keys(['entryType', 'primaryAccession', 'uniProtkbId', 'entryAudit', 'annotationScore', 'organism', 'proteinExistence', 'proteinDescription', 'genes', 'comments', 'features', 'keywords', 'references', 'uniProtKBCrossReferences', 'sequence', 'extraAttributes'])

I need a for cycle which will go through all proteins. There I'm going to look at primaryAccession where the Uniprot ID of the protein is. I'm also going to look at key 'comments' where we can find all the information abou the reaction.

In [8]:
print(len(protein_access))

1726


We have 1726 reviewed entries in this json file

In [15]:
data = []

for i in range(len(protein_access)):
    id = protein_access[i]['primaryAccession']
    ptDescription = protein_access[i]['proteinDescription']
    name = ptDescription['recommendedName']['fullName']['value']
    comments = protein_access[i]['comments']
    reaction_found = False
    cofactor_found = False

    for j in range(len(comments)):
        if comments[j]['commentType'] == 'COFACTOR':
            cofactors = comments[j]['cofactors']
            cofactor = cofactors[0]['name']
            #print(cofactor)
        if comments[j]['commentType'] == "CATALYTIC ACTIVITY":
            reaction = comments[j]['reaction']['name']
            reactionSplit = reaction.split(' = ')
            substrates = reactionSplit[0].strip()
            products = reactionSplit[1].strip()
            
            substratesSplit = substrates.split(' + ')
            productsSplit = products.split(' + ')
            participants = substratesSplit + productsSplit
            
            reducing_agent = '' 
            new_substrates = []
            new_products = []
            for sub in productsSplit:
                if sub.strip() not in ['H(+)', 'H2O', 'AH2', '2 H(+)', '2 H2O','3 H(+)', '3 H2O', '4 H(+)', '4 H2O', '6 H2O', 'oxidized [NADPH--hemoprotein reductase]', 
                           'oxidized [2Fe-2S]-[ferredoxin]', 'oxidized [adrenodoxin]', '2 oxidized [NADPH--hemoprotein reductase]', '6 oxidized [NADPH--hemoprotein reductase]', 
                           '2 oxidized [2Fe-2S]-[ferredoxin]', '2 oxidized [adrenodoxin]', '6 oxidized [2Fe-2S]-[ferredoxin]', '6 oxidized [adrenodoxin]',
                           'A', 'NAD(+)', 'NADP(+)', 'NADPH', 'NADH',  '2 NADP(+)', '2 NADPH', '2 NADH', 'CO2', '2 CO2', '3 CO2',  '3 oxidized [NADPH--hemoprotein reductase]', 
                           '3 oxidized [2Fe-2S]-[ferredoxin]', '3 oxidized [adrenodoxin]', '4 oxidized [NADPH--hemoprotein reductase]', 
                           '4 oxidized [2Fe-2S]-[ferredoxin]', '4 oxidized [adrenodoxin]', 'oxidized [NADH--hemoprotein reductase]', '2 oxidized [flavodoxin]',  'oxidized 2[4Fe-4S]-[ferredoxin]']:
                    new_products.append(sub.strip())
            
            for sub in substratesSplit:
                if 'reduced [' in sub:
                    reducing_agent = sub.strip()[0:9] + sub.strip()[9:-1] + sub.strip()[-1:]
                elif '2 reduced ferredoxin [iron-sulfur] cluster' in sub:
                    reducing_agent =  '2 reduced ferredoxin [iron-sulfur] cluster'
                elif 'reduced 2[4Fe-4S]-[ferredoxin]' in sub:
                    reducing_agent =  'reduced 2[4Fe-4S]-[ferredoxin]'
                elif sub in ['NADPH', 'NADP(+)', 'NAD(+)', 'NADH', '2 NADPH', '3 NADPH']:
                    reducing_agent= sub.strip()
                elif sub not in ['O2', 'H(+)', 'AH2', 'H2O', 'H2O2', 'A', '2 O2', '2 H(+)', '2 AH2', '2 H2O', '2 H2O2', 
                                 '2 A', '3 O2', '3 H(+)', '3 AH2', '3 H2O', '3 H2O2', '3 A', '4 H(+)', '5 H(+)', '6 H(+)', '4 O2']:
                    new_substrates.append(sub.strip())

                    
            crossRef = comments[j]['reaction'].get('reactionCrossReferences', [])
            
            rheaID = ''
            chebis = []
            participant_chebis = {}
            for ref in crossRef:
                if ref['database'] == 'Rhea':
                    rheaID = ref['id']
                    break
                     
            
            data.append({'protein': id, 'name': name, 'reaction': reaction, 'rheaID': rheaID, 'substrates': new_substrates, 
                         'Substrate CHEBI ID': None, 'Substrate SMILES': None, 'reducing_agent': reducing_agent, 'cofactor': cofactor,
                         'products': new_products, 'Product CHEBI ID': None, 'Product SMILES': None, 
                         'participants': participants})
            reaction_found = True
            cofactor_found = True


    # before adding 2929 records in dataframe, after adding this 3781 records - so we have 2929 fully annotated reactions from reviewed 'PF00067'
    if not reaction_found and not cofactor_found:
        data.append({'protein': id, 'name': name, 'reaction': None, 'rheaID': None, 'reducing_agent': None,'cofactor': None, 'substrates': None, 
                         'products': None,'participants': None})

df = pd.DataFrame(data)

df.to_csv('reactionsfromjson.csv', index=False)
#df.to_excel('reactionFromJSONexcel.xlsx', index=False)

Now we need to add ChEBI IDs & SMILES of the substrates and the products to the dataframe from reaction_data.csv file.

In [17]:
reaction_data = pd.read_csv("reaction_data.csv")
reaction_data.head(20)

/tmp/ipykernel_31446/4129889877.py:1: DtypeWarning: Columns (1) have mixed types. Specify dtype option on import or set low_memory=False.
  reaction_data = pd.read_csv("reaction_data.csv")


protein reaction  reaction_side_order         chebi  \
0   Q7Z1V1    54028                    1  CHEBI_138029   
1   Q7Z1V1    54028                    1   CHEBI_15379   
2   Q7Z1V1    54028                    1   CHEBI_57618   
3   Q7Z1V1    54028                    2  CHEBI_138031   
4   Q7Z1V1    54028                    2   CHEBI_15740   
5   Q7Z1V1    54028                    2   CHEBI_15378   
6   Q7Z1V1    54028                    2   CHEBI_15377   
7   Q7Z1V1    54028                    2   CHEBI_58210   
8   Q9LJK2    12897                    1   CHEBI_37569   
9   Q9LJK2    12897                    1   CHEBI_15379   
10  Q9LJK2    12897                    1   CHEBI_57618   
11  Q9LJK2    12897                    2   CHEBI_58490   
12  Q9LJK2    12897                    2   CHEBI_15378   
13  Q9LJK2    12897                    2   CHEBI_15377   
14  Q9LJK2    12897                    2   CHEBI_58210   
15  Q949P1    12897                    1   CHEBI_37569   
16  Q949P1    12897                    1   CHEBI_15379   
17  Q949P1    12897                    1   CHEBI_57618   
18  Q949P1    12897                    2   CHEBI_58490   
19  Q949P1    12897                    2   CHEBI_15378   

                                               smiles  
0        C12C([C@]3(C(C(CC3)*)(C)CC1)C)CCC4C2(CCCC4)C  
1                                                 O=O  
2   Cc1cc2Nc3c([nH]c(=O)[nH]c3=O)N(C[C@H](O)[C@H](...  
3             C12C(C=3C(C(CC3)*)(C)CC1)CCC4C2(CCCC4)C  
4                                        [H]C([O-])=O  
5                                                [H+]  
6                                             [H]O[H]  
7   C12=NC([N-]C(C1=NC=3C(N2C[C@@H]([C@@H]([C@@H](...  
8    CC(\C=C\[C@@]1(O)C(C)=CC(=O)CC1(C)C)=C\C([O-])=O  
9                                                 O=O  
10  Cc1cc2Nc3c([nH]c(=O)[nH]c3=O)N(C[C@H](O)[C@H](...  
11  CC(\C=C\[C@@]1(O)C(C)=CC(=O)C[C@]1(C)CO)=C\C([...  
12                                               [H+]  
13                                            [H]O[H]  
14  C12=NC([N-]C(C1=NC=3C(N2C[C@@H]([C@@H]([C@@H](...  
15   CC(\C=C\[C@@]1(O)C(C)=CC(=O)CC1(C)C)=C\C([O-])=O  
16                                                O=O  
17  Cc1cc2Nc3c([nH]c(=O)[nH]c3=O)N(C[C@H](O)[C@H](...  
18  CC(\C=C\[C@@]1(O)C(C)=CC(=O)C[C@]1(C)CO)=C\C([...  
19                                               [H+]

In [18]:
#getting rid of 'RHEA:' in my df to simplify the search of ChEBI IDs and SMILES in reaction_data.csv file
df['rheaID'] = df['rheaID'].str.replace("RHEA:","")

protein                                     name  \
0   A0A075TMP8       Cytochrome P450 monooxygenase patI   
1   A0A075TRL5       Cytochrome P450 monooxygenase patH   
2   A0A087X1C5             Putative cytochrome P450 2D7   
3   A0A0C5Q4Y6                    Ferruginol synthase 1   
4   A0A0C5Q4Y6                    Ferruginol synthase 1   
5   A0A0C5Q4Y6                    Ferruginol synthase 1   
6   A0A0C5QRZ2                      Ferruginol synthase   
7   A0A0C5QRZ2                      Ferruginol synthase   
8   A0A0C5QRZ2                      Ferruginol synthase   
9   A0A1D6HSP4              Dimethylnonatriene synthase   
10  A0A1D6HSP4              Dimethylnonatriene synthase   
11  A0A2H4DGV8     Germacrene A acid 8-beta-hydroxylase   
12  A0A2H4DGV8     Germacrene A acid 8-beta-hydroxylase   
13  A0A2H4DGV8     Germacrene A acid 8-beta-hydroxylase   
14  A0A2Z5D854                     Xanthotoxol synthase   
15  A0A2Z5D854                     Xanthotoxol synthase   
16  A0A343URW6      Tabersonine 6,7-epoxidase isoform 1   
17  A0A517FNB9  Cholesterol 22-monohydroxylase CYP90B52   
18  A0A517FNC5  Cholesterol 16,22-dihydroxylase CYP90G4   
19  A0A517FNC6  Cholesterol 22-monohydroxylase CYP90B51   

                                             reaction rheaID  \
0   3-hydroxybenzyl alcohol + O2 + reduced [NADPH-...  62212   
1   3-methylphenol + O2 + reduced [NADPH--hemoprot...  62208   
2   an organic molecule + O2 + reduced [NADPH--hem...  17149   
3   abieta-8,11,13-triene + O2 + reduced [NADPH--h...  48080   
4   ferruginol + O2 + reduced [NADPH--hemoprotein ...  55428   
5   miltiradiene + 2 O2 + 2 reduced [NADPH--hemopr...  66796   
6   abieta-8,11,13-triene + O2 + reduced [NADPH--h...  48080   
7   ferruginol + O2 + reduced [NADPH--hemoprotein ...  55428   
8   miltiradiene + 2 O2 + 2 reduced [NADPH--hemopr...  66796   
9   (6E,10E)-geranyllinalool + O2 + reduced [NADPH...  13545   
10  (3S,6E)-nerolidol + O2 + reduced [NADPH--hemop...  55424   
11  germacra-1(10),4,11(13)-trien-12-oate + O2 + r...  57964   
12  germacra-1(10),4,11(13)-trien-12-oate + O2 + r...  57968   
13  germacra-1(10),4,11(13)-trien-12-oate + O2 + r...  58032   
14  O2 + psoralen + reduced [NADPH--hemoprotein re...  68548   
15  6-methoxycoumarin + O2 + reduced [NADPH--hemop...  68564   
16  (-)-tabersonine + O2 + reduced [NADPH--hemopro...  61056   
17  cholesterol + O2 + reduced [NADPH--hemoprotein...  69839   
18  cholesterol + 2 O2 + 2 reduced [NADPH--hemopro...  72191   
19  cholesterol + O2 + reduced [NADPH--hemoprotein...  69839   

                                 substrates  Substrate CHEBI ID  \
0                 [3-hydroxybenzyl alcohol]                 NaN   
1                          [3-methylphenol]                 NaN   
2                     [an organic molecule]                 NaN   
3                   [abieta-8,11,13-triene]                 NaN   
4                              [ferruginol]                 NaN   
5                            [miltiradiene]                 NaN   
6                   [abieta-8,11,13-triene]                 NaN   
7                              [ferruginol]                 NaN   
8                            [miltiradiene]                 NaN   
9                [(6E,10E)-geranyllinalool]                 NaN   
10                      [(3S,6E)-nerolidol]                 NaN   
11  [germacra-1(10),4,11(13)-trien-12-oate]                 NaN   
12  [germacra-1(10),4,11(13)-trien-12-oate]                 NaN   
13  [germacra-1(10),4,11(13)-trien-12-oate]                 NaN   
14                               [psoralen]                 NaN   
15                      [6-methoxycoumarin]                 NaN   
16                        [(-)-tabersonine]                 NaN   
17                            [cholesterol]                 NaN   
18                            [cholesterol]                 NaN   
19                            [cholesterol]                 NaN   

    Substrate 

Create a dictionary from the reaction_data dataframe, where tuple protein, reaction will be the key and value will be the array of chebis and smiles

In [19]:
# creating dictionary from reaction_data where key = (protein, reaction), value = list of remaining columns
protein_reaction = reaction_data.groupby(['protein', 'reaction'])
reaction_dict = {(k[0], k[1]): [list(row[3:]) for _, row in g.iterrows()] for k, g in protein_reaction}

In [20]:
# just an example of the values for key ("Q7Z1V1", "54028")
print(reaction_dict[("Q7Z1V1","54028")])

[['CHEBI_138029', 'C12C([C@]3(C(C(CC3)*)(C)CC1)C)CCC4C2(CCCC4)C'], ['CHEBI_15379', 'O=O'], ['CHEBI_57618', 'Cc1cc2Nc3c([nH]c(=O)[nH]c3=O)N(C[C@H](O)[C@H](O)[C@H](O)COP([O-])([O-])=O)c2cc1C'], ['CHEBI_138031', 'C12C(C=3C(C(CC3)*)(C)CC1)CCC4C2(CCCC4)C'], ['CHEBI_15740', '[H]C([O-])=O'], ['CHEBI_15378', '[H+]'], ['CHEBI_15377', '[H]O[H]'], ['CHEBI_58210', 'C12=NC([N-]C(C1=NC=3C(N2C[C@@H]([C@@H]([C@@H](COP(=O)([O-])[O-])O)O)O)=CC(=C(C3)C)C)=O)=O']]


In [21]:
# iterate through each row of df - find the tuple (id,rheaID) in the reaction_dict
# also get the index of each substrate and the index of each product in the participants list 
for index, row in df.iterrows():
    substrate_indexes = []
    product_indexes = []
    if row['substrates'] and row['products']:
        for subs in row['substrates']:
            if subs in row['participants']:
                substrate_indexes.append(row['participants'].index(subs))
        for prod in row['products']:
            if prod in row['participants']:
                product_indexes.append(row['participants'].index(prod))
    idXrheaID = (row['protein'], row['rheaID'])
    #print(idXrheaID,substrate_indexes, product_indexes)

    # now we need to find ChEBI ID & SMILES in the reaction_dict based on the tuple & indexes we got here

    # retrieve the list of remaining columns for the idXrheaID from the reaction_dict
    if idXrheaID in reaction_dict:
        all_vals = reaction_dict[idXrheaID]
        maxi = len(all_vals) - 1

        # iterate through the substrate_indexes and product_indexes lists to retrieve the corresponding values
        substrates = [all_vals[i] for i in substrate_indexes if i <= maxi]
        products = [all_vals[i] for i in product_indexes if i <= maxi]

        substrate_ids = [s[0] for s in substrates]
        substrate_smiles = [s[1] for s in substrates]
        product_ids = [p[0] for p in products]
        product_smiles = [p[1] for p in products]

        df.loc[index, 'Substrate CHEBI ID'] = ";".join(substrate_ids)
        df.loc[index, 'Substrate SMILES'] = ";".join(substrate_smiles)
        df.loc[index, 'Product CHEBI ID'] = ";".join(product_ids)
        df.loc[index, 'Product SMILES'] = ";".join(product_smiles)
    
    

df['products'] = df['products'].apply(lambda x: ';'.join(x) if isinstance(x,list) else x)
df['substrates'] = df['substrates'].apply(lambda x: ';'.join(x) if isinstance(x,list) else x)

df.head(20)

        

protein                                     name  \
0   A0A075TMP8       Cytochrome P450 monooxygenase patI   
1   A0A075TRL5       Cytochrome P450 monooxygenase patH   
2   A0A087X1C5             Putative cytochrome P450 2D7   
3   A0A0C5Q4Y6                    Ferruginol synthase 1   
4   A0A0C5Q4Y6                    Ferruginol synthase 1   
5   A0A0C5Q4Y6                    Ferruginol synthase 1   
6   A0A0C5QRZ2                      Ferruginol synthase   
7   A0A0C5QRZ2                      Ferruginol synthase   
8   A0A0C5QRZ2                      Ferruginol synthase   
9   A0A1D6HSP4              Dimethylnonatriene synthase   
10  A0A1D6HSP4              Dimethylnonatriene synthase   
11  A0A2H4DGV8     Germacrene A acid 8-beta-hydroxylase   
12  A0A2H4DGV8     Germacrene A acid 8-beta-hydroxylase   
13  A0A2H4DGV8     Germacrene A acid 8-beta-hydroxylase   
14  A0A2Z5D854                     Xanthotoxol synthase   
15  A0A2Z5D854                     Xanthotoxol synthase   
16  A0A343URW6      Tabersonine 6,7-epoxidase isoform 1   
17  A0A517FNB9  Cholesterol 22-monohydroxylase CYP90B52   
18  A0A517FNC5  Cholesterol 16,22-dihydroxylase CYP90G4   
19  A0A517FNC6  Cholesterol 22-monohydroxylase CYP90B51   

                                             reaction rheaID  \
0   3-hydroxybenzyl alcohol + O2 + reduced [NADPH-...  62212   
1   3-methylphenol + O2 + reduced [NADPH--hemoprot...  62208   
2   an organic molecule + O2 + reduced [NADPH--hem...  17149   
3   abieta-8,11,13-triene + O2 + reduced [NADPH--h...  48080   
4   ferruginol + O2 + reduced [NADPH--hemoprotein ...  55428   
5   miltiradiene + 2 O2 + 2 reduced [NADPH--hemopr...  66796   
6   abieta-8,11,13-triene + O2 + reduced [NADPH--h...  48080   
7   ferruginol + O2 + reduced [NADPH--hemoprotein ...  55428   
8   miltiradiene + 2 O2 + 2 reduced [NADPH--hemopr...  66796   
9   (6E,10E)-geranyllinalool + O2 + reduced [NADPH...  13545   
10  (3S,6E)-nerolidol + O2 + reduced [NADPH--hemop...  55424   
11  germacra-1(10),4,11(13)-trien-12-oate + O2 + r...  57964   
12  germacra-1(10),4,11(13)-trien-12-oate + O2 + r...  57968   
13  germacra-1(10),4,11(13)-trien-12-oate + O2 + r...  58032   
14  O2 + psoralen + reduced [NADPH--hemoprotein re...  68548   
15  6-methoxycoumarin + O2 + reduced [NADPH--hemop...  68564   
16  (-)-tabersonine + O2 + reduced [NADPH--hemopro...  61056   
17  cholesterol + O2 + reduced [NADPH--hemoprotein...  69839   
18  cholesterol + 2 O2 + 2 reduced [NADPH--hemopro...  72191   
19  cholesterol + O2 + reduced [NADPH--hemoprotein...  69839   

                               substrates Substrate CHEBI ID  \
0                 3-hydroxybenzyl alcohol        CHEBI_17069   
1                          3-methylphenol        CHEBI_17231   
2                     an organic molecule       CHEBI_142491   
3                   abieta-8,11,13-triene        CHEBI_86062   
4                              ferruginol        CHEBI_78274   
5                            miltiradiene        CHEBI_65037   
6                   abieta-8,11,13-triene        CHEBI_86062   
7                              ferruginol        CHEBI_78274   
8                            miltiradiene        CHEBI_65037   
9                (6E,10E)-geranyllinalool        CHEBI_74299   
10                      (3S,6E)-nerolidol        CHEBI_59958   
11  germacra-1(10),4,11(13)-trien-12-oate        CHEBI_61301   
12  germacra-1(10),4,11(13)-trien-12-oate        CHEBI_61301   
13  germacra-1(10),4,11(13)-trien-12-oate        CHEBI_61301   
14                               psoralen        CHEBI_27616   
15                      6-methoxycoumarin       CHEBI_178005   
16                        (-)-tabersonine        CHEBI_57893   
17                            cholesterol        CHEBI_16113   
18                            cholesterol        CHEBI_16113   
19                            cholesterol        CHEBI_16113   

                                     Substrate SMILES  \
0                   

I have a dataframe containing Uniprot ID,name, reaction, RHEA ID, substrates,
Substrate CHEBI ID, Substrate SMILES, reducing agent, cofactor, products, Product CHEBI ID, Product SMILES & the list of participants which I don't need anymore (but I'm still going to keep it for now).

Next things I want to do: add species & kingdom from taxa.csv, add sequences, lenghts (I have these in my google sheets db).

I'm missing chemical formula of product here - decide on how to add it to the database

In [22]:
taxa = pd.read_csv("./taxaNAT.csv",index_col='protein',sep=',')
taxa = taxa.fillna("MISSING VALUE")

taxa.head(20)

organism_name   kingdom_name  \
protein                                                               
A0A1H9YG21                  Nitrosospira multiformis  MISSING VALUE   
A0A1H8EWZ4                  Nitrosospira multiformis  MISSING VALUE   
A0A1I7H525                  Nitrosospira multiformis  MISSING VALUE   
A0A2T5IGS8                  Nitrosospira multiformis  MISSING VALUE   
Q577H8      Brucella abortus biovar 1 (strain 9-941)  MISSING VALUE   
A0A4X2K8V6                          Vombatus ursinus        Metazoa   
A0A4X2MFM9                          Vombatus ursinus        Metazoa   
A0A4X2KZM1                          Vombatus ursinus        Metazoa   
A0A4X2KZT1                          Vombatus ursinus        Metazoa   
A0A4X2LU16                          Vombatus ursinus        Metazoa   
A0A4X2JPW9                          Vombatus ursinus        Metazoa   
A0A4X2K4C2                          Vombatus ursinus        Metazoa   
A0A4X2LJ74                          Vombatus ursinus        Metazoa   
A0A4X2MEW6                          Vombatus ursinus        Metazoa   
A0A4X2LSE6                          Vombatus ursinus        Metazoa   
A0A4X2KPL9                          Vombatus ursinus        Metazoa   
A0A4X2LJ98                          Vombatus ursinus        Metazoa   
A0A4X2M4E4                          Vombatus ursinus        Metazoa   
A0A4X2JX08                          Vombatus ursinus        Metazoa   
A0A4X2MBM2                          Vombatus ursinus        Metazoa   

           superkingdom_name  
protein                       
A0A1H9YG21          Bacteria  
A0A1H8EWZ4          Bacteria  
A0A1I7H525          Bacteria  
A0A2T5IGS8          Bacteria  
Q577H8              Bacteria  
A0A4X2K8V6         Eukaryota  
A0A4X2MFM9         Eukaryota  
A0A4X2KZM1         Eukaryota  
A0A4X2KZT1         Eukaryota  
A0A4X2LU16         Eukaryota  
A0A4X2JPW9         Eukaryota  
A0A4X2K4C2         Eukaryota  
A0A4X2LJ74         Eukaryota  
A0A4X2MEW6         Eukaryota  
A0A4X2LSE6         Eukaryota  
A0A4X2KPL9         Eukaryota  
A0A4X2LJ98         Eukaryota  
A0A4X2M4E4         Eukaryota  
A0A4X2JX08         Eukaryota  
A0A4X2MBM2         Eukaryota

In [23]:
taxa_sp_king = taxa.drop(columns=['superkingdom_name'])

In [24]:
merged_df = pd.merge(df, taxa, left_on='protein', right_index=True)
merged_df.head(20)


protein                                     name  \
0   A0A075TMP8       Cytochrome P450 monooxygenase patI   
1   A0A075TRL5       Cytochrome P450 monooxygenase patH   
2   A0A087X1C5             Putative cytochrome P450 2D7   
3   A0A0C5Q4Y6                    Ferruginol synthase 1   
4   A0A0C5Q4Y6                    Ferruginol synthase 1   
5   A0A0C5Q4Y6                    Ferruginol synthase 1   
6   A0A0C5QRZ2                      Ferruginol synthase   
7   A0A0C5QRZ2                      Ferruginol synthase   
8   A0A0C5QRZ2                      Ferruginol synthase   
9   A0A1D6HSP4              Dimethylnonatriene synthase   
10  A0A1D6HSP4              Dimethylnonatriene synthase   
11  A0A2H4DGV8     Germacrene A acid 8-beta-hydroxylase   
12  A0A2H4DGV8     Germacrene A acid 8-beta-hydroxylase   
13  A0A2H4DGV8     Germacrene A acid 8-beta-hydroxylase   
14  A0A2Z5D854                     Xanthotoxol synthase   
15  A0A2Z5D854                     Xanthotoxol synthase   
16  A0A343URW6      Tabersonine 6,7-epoxidase isoform 1   
17  A0A517FNB9  Cholesterol 22-monohydroxylase CYP90B52   
18  A0A517FNC5  Cholesterol 16,22-dihydroxylase CYP90G4   
19  A0A517FNC6  Cholesterol 22-monohydroxylase CYP90B51   

                                             reaction rheaID  \
0   3-hydroxybenzyl alcohol + O2 + reduced [NADPH-...  62212   
1   3-methylphenol + O2 + reduced [NADPH--hemoprot...  62208   
2   an organic molecule + O2 + reduced [NADPH--hem...  17149   
3   abieta-8,11,13-triene + O2 + reduced [NADPH--h...  48080   
4   ferruginol + O2 + reduced [NADPH--hemoprotein ...  55428   
5   miltiradiene + 2 O2 + 2 reduced [NADPH--hemopr...  66796   
6   abieta-8,11,13-triene + O2 + reduced [NADPH--h...  48080   
7   ferruginol + O2 + reduced [NADPH--hemoprotein ...  55428   
8   miltiradiene + 2 O2 + 2 reduced [NADPH--hemopr...  66796   
9   (6E,10E)-geranyllinalool + O2 + reduced [NADPH...  13545   
10  (3S,6E)-nerolidol + O2 + reduced [NADPH--hemop...  55424   
11  germacra-1(10),4,11(13)-trien-12-oate + O2 + r...  57964   
12  germacra-1(10),4,11(13)-trien-12-oate + O2 + r...  57968   
13  germacra-1(10),4,11(13)-trien-12-oate + O2 + r...  58032   
14  O2 + psoralen + reduced [NADPH--hemoprotein re...  68548   
15  6-methoxycoumarin + O2 + reduced [NADPH--hemop...  68564   
16  (-)-tabersonine + O2 + reduced [NADPH--hemopro...  61056   
17  cholesterol + O2 + reduced [NADPH--hemoprotein...  69839   
18  cholesterol + 2 O2 + 2 reduced [NADPH--hemopro...  72191   
19  cholesterol + O2 + reduced [NADPH--hemoprotein...  69839   

                               substrates Substrate CHEBI ID  \
0                 3-hydroxybenzyl alcohol        CHEBI_17069   
1                          3-methylphenol        CHEBI_17231   
2                     an organic molecule       CHEBI_142491   
3                   abieta-8,11,13-triene        CHEBI_86062   
4                              ferruginol        CHEBI_78274   
5                            miltiradiene        CHEBI_65037   
6                   abieta-8,11,13-triene        CHEBI_86062   
7                              ferruginol        CHEBI_78274   
8                            miltiradiene        CHEBI_65037   
9                (6E,10E)-geranyllinalool        CHEBI_74299   
10                      (3S,6E)-nerolidol        CHEBI_59958   
11  germacra-1(10),4,11(13)-trien-12-oate        CHEBI_61301   
12  germacra-1(10),4,11(13)-trien-12-oate        CHEBI_61301   
13  germacra-1(10),4,11(13)-trien-12-oate        CHEBI_61301   
14                               psoralen        CHEBI_27616   
15                      6-methoxycoumarin       CHEBI_178005   
16                        (-)-tabersonine        CHEBI_57893   
17                            cholesterol        CHEBI_16113   
18                            cholesterol        CHEBI_16113   
19                            cholesterol        CHEBI_16113   

                                     Substrate SMILES  \
0                   

In [25]:
merged_df.columns

Index(['protein', 'name', 'reaction', 'rheaID', 'substrates',
       'Substrate CHEBI ID', 'Substrate SMILES', 'reducing_agent', 'cofactor',
       'products', 'Product CHEBI ID', 'Product SMILES', 'participants',
       'organism_name', 'kingdom_name', 'superkingdom_name'],
      dtype='object')

In [26]:
kingdom_counts = merged_df.groupby('kingdom_name').count().sort_values('protein', ascending = False)

In [27]:
kingdom_counts.head(30)

protein  name  reaction  rheaID  substrates  \
kingdom_name                                                 
Metazoa           2239  2239      2117    2117        2117   
Viridiplantae      705   705       505     505         505   
Fungi              586   586       177     177         177   
MISSING VALUE      248   248       128     128         128   
Bamfordvirae         2     2         1       1           1   

               Substrate CHEBI ID  Substrate SMILES  reducing_agent  cofactor  \
kingdom_name                                                                    
Metazoa                      2117              2117            2117      2117   
Viridiplantae                 505               505             505       505   
Fungi                         177               177             177       177   
MISSING VALUE                 125               125             128       128   
Bamfordvirae                    1                 1               1         1   

               products  Product CHEBI ID  Product SMILES  participants  \
kingdom_name                                                              
Metazoa            2117              2117            2117          2117   
Viridiplantae       505               505             505           505   
Fungi               177               177             177           177   
MISSING VALUE       128               125             125           128   
Bamfordvirae          1                 1               1             1   

               organism_name  superkingdom_name  
kingdom_name                                     
Metazoa                 2239               2239  
Viridiplantae            705                705  
Fungi                    586                586  
MISSING VALUE            248                248  
Bamfordvirae               2                  2

Metazoa = Animalia</br>
Viridiplantae = "green plantae" = Plantae</br>
Bamfordvirae = viruses</br>
Bacteria are in MISSING VALUE</br>


In [28]:
merged_df['kingdom_name'] = merged_df['kingdom_name'].str.replace("Metazoa","Animalia")
merged_df['kingdom_name'] = merged_df['kingdom_name'].str.replace("Viridiplantae","Plantae")
mask = (merged_df['kingdom_name'] == 'MISSING VALUE') & (merged_df['superkingdom_name'] == 'Bacteria')
merged_df.loc[mask, 'kingdom_name'] = 'Bacteria'

In [29]:
pfams = pd.read_csv("./pfams.csv", index_col='protein', sep=',')
pfams = pfams.fillna("MISSING VALUE")

pfams.head(20)

pfam
protein            
A0A5N4AF60  PF00009
A0A3M7M744  PF00009
A0A8J5EMQ6  PF00009
A0A8H7BGB7  PF00009
A0A8H7PNS5  PF00009
A0A364NAN5  PF00009
A0A3M0WC62  PF00009
A0A7R9BNE8  PF00009
A0A3M7MT89  PF00009
A0A8J5EMQ6  PF03143
A0A8J5EMQ6  PF03144
A0A8H7PNS5  PF03144
A0A7R9BNE8  PF03144
A0A2I0JIH5  PF00533
A0A7J6FAH0  PF01016
A0A498HIC2  PF01016
A0A7J6HQF5  PF01016
A0A3L6DA02  PF01470
A0A0J8BE80  PF02190
A0A524JHM1  PF03726

In [30]:
merged_df_with_pfams = pd.merge(merged_df, pfams, left_on='protein', right_index=True)
merged_df_with_pfams.head(20)


protein                                     name  \
0   A0A075TMP8       Cytochrome P450 monooxygenase patI   
1   A0A075TRL5       Cytochrome P450 monooxygenase patH   
2   A0A087X1C5             Putative cytochrome P450 2D7   
3   A0A0C5Q4Y6                    Ferruginol synthase 1   
4   A0A0C5Q4Y6                    Ferruginol synthase 1   
5   A0A0C5Q4Y6                    Ferruginol synthase 1   
6   A0A0C5QRZ2                      Ferruginol synthase   
7   A0A0C5QRZ2                      Ferruginol synthase   
8   A0A0C5QRZ2                      Ferruginol synthase   
9   A0A1D6HSP4              Dimethylnonatriene synthase   
10  A0A1D6HSP4              Dimethylnonatriene synthase   
11  A0A2H4DGV8     Germacrene A acid 8-beta-hydroxylase   
12  A0A2H4DGV8     Germacrene A acid 8-beta-hydroxylase   
13  A0A2H4DGV8     Germacrene A acid 8-beta-hydroxylase   
14  A0A2Z5D854                     Xanthotoxol synthase   
15  A0A2Z5D854                     Xanthotoxol synthase   
16  A0A343URW6      Tabersonine 6,7-epoxidase isoform 1   
17  A0A517FNB9  Cholesterol 22-monohydroxylase CYP90B52   
18  A0A517FNC5  Cholesterol 16,22-dihydroxylase CYP90G4   
19  A0A517FNC6  Cholesterol 22-monohydroxylase CYP90B51   

                                             reaction rheaID  \
0   3-hydroxybenzyl alcohol + O2 + reduced [NADPH-...  62212   
1   3-methylphenol + O2 + reduced [NADPH--hemoprot...  62208   
2   an organic molecule + O2 + reduced [NADPH--hem...  17149   
3   abieta-8,11,13-triene + O2 + reduced [NADPH--h...  48080   
4   ferruginol + O2 + reduced [NADPH--hemoprotein ...  55428   
5   miltiradiene + 2 O2 + 2 reduced [NADPH--hemopr...  66796   
6   abieta-8,11,13-triene + O2 + reduced [NADPH--h...  48080   
7   ferruginol + O2 + reduced [NADPH--hemoprotein ...  55428   
8   miltiradiene + 2 O2 + 2 reduced [NADPH--hemopr...  66796   
9   (6E,10E)-geranyllinalool + O2 + reduced [NADPH...  13545   
10  (3S,6E)-nerolidol + O2 + reduced [NADPH--hemop...  55424   
11  germacra-1(10),4,11(13)-trien-12-oate + O2 + r...  57964   
12  germacra-1(10),4,11(13)-trien-12-oate + O2 + r...  57968   
13  germacra-1(10),4,11(13)-trien-12-oate + O2 + r...  58032   
14  O2 + psoralen + reduced [NADPH--hemoprotein re...  68548   
15  6-methoxycoumarin + O2 + reduced [NADPH--hemop...  68564   
16  (-)-tabersonine + O2 + reduced [NADPH--hemopro...  61056   
17  cholesterol + O2 + reduced [NADPH--hemoprotein...  69839   
18  cholesterol + 2 O2 + 2 reduced [NADPH--hemopro...  72191   
19  cholesterol + O2 + reduced [NADPH--hemoprotein...  69839   

                               substrates Substrate CHEBI ID  \
0                 3-hydroxybenzyl alcohol        CHEBI_17069   
1                          3-methylphenol        CHEBI_17231   
2                     an organic molecule       CHEBI_142491   
3                   abieta-8,11,13-triene        CHEBI_86062   
4                              ferruginol        CHEBI_78274   
5                            miltiradiene        CHEBI_65037   
6                   abieta-8,11,13-triene        CHEBI_86062   
7                              ferruginol        CHEBI_78274   
8                            miltiradiene        CHEBI_65037   
9                (6E,10E)-geranyllinalool        CHEBI_74299   
10                      (3S,6E)-nerolidol        CHEBI_59958   
11  germacra-1(10),4,11(13)-trien-12-oate        CHEBI_61301   
12  germacra-1(10),4,11(13)-trien-12-oate        CHEBI_61301   
13  germacra-1(10),4,11(13)-trien-12-oate        CHEBI_61301   
14                               psoralen        CHEBI_27616   
15                      6-methoxycoumarin       CHEBI_178005   
16                        (-)-tabersonine        CHEBI_57893   
17                            cholesterol        CHEBI_16113   
18                            cholesterol        CHEBI_16113   
19                            cholesterol        CHEBI_16113   

                                     Substrate SMILES  \
0                   

In [31]:
seqs = pd.read_csv("./sequencesNAT.csv",index_col='protein',sep=',')
seqs = seqs.fillna("MISSING VALUE")

seqs.head(20)

sequence
protein                                                   
P05108   MLAKGLPPRSVLVKGCQTFLSAPREGLGRLRVPTGEGAGISTRSPR...
Q9V7G5   MFVEKVLERTTNLELCSILILLVISLSIYTFYATLNTYLRSVLLSL...
O65815   MEIPYLLTTTLLLLFTTLYLLLRRRSSTLPPTIFPSLPIIGHLYLL...
Q9FMA5   MGAMMVMMGLLLIIVSLCSALLRWNQMRYTKNGLPPGTMGWPIFGE...
P08686   MLLLGLLLLPLLAGARLLWNWWKLRSLHLPPLAPGFLHLLQPDLPI...
P24557   MEALGFLKLEVNGPMVTVALSVALLALLKWYSTSAFSRLEKLGLRH...
Q9T0K2   MEMILITLCLTTLLALLLKSILKRTATKNFNLPPSPWRLPVIGNLH...
Q9LXM3   MSIFLCFLLLLPLSLILFKKLLPSKGKLPPGPIGLPIIGNLHQLGK...
Q9CX98   MSSLGDQRPAAGEQPGARLHVRATGGALLLCLLAVLLGWVWLRRQR...
Q9VGZ0   MLSTQWNANKQISRQIYQLCRGLAQKATAVNLEEAKPYADIPGPSK...
Q96SQ9   MEATGTWALLLALALLLLLTLALSGTRARGHLPPGPTPLPLLGNLL...
P11511   MVLEMLNPIHYNITSIVPEAMPAATMPVLLLTGLFLLVWNYEGTSS...
Q8N118   MEFSWLETRWARPFYLAFVFCLALGLLQAIKLYLRRQRLLRDLRPF...
Q91X77   MDPILVLVFTLSCLFLLSLWRQSSERGKLPPGPTPLPIIGNILQIN...
Q9LSF8   MTFLFSTLQLSLFSLALVIFGYIFLRKQLSRCEVDSSTIPEPLGAL...
Q08477   MPQLSLSSLGLWPMAASPWLLLLLVGASWLLARILAWTYTFYDNCC...
P33269   MFLVIGAILASALFVGLLLYHLKFKRLIDLISYMPGPPVLPLVGHG...
Q9HCS2   MSLLSLPWLGLRPVATSPWLLLLLVVGSWLLARILAWTYAFYNNCR...
O64636   MDIISGQALLLLFCFILSCFLIFTTTRSGRISRGATALPPGPPRLP...
P10635   MGLEALVPLAVIVAIFLLLVDLMHRRQRWAARYPPGPLPLPGLGNL...

In [32]:
merged_df_with_seqs = pd.merge(merged_df_with_pfams, seqs, left_on='protein', right_index=True)
merged_df_with_seqs.head(20)


protein                                     name  \
0   A0A075TMP8       Cytochrome P450 monooxygenase patI   
1   A0A075TRL5       Cytochrome P450 monooxygenase patH   
2   A0A087X1C5             Putative cytochrome P450 2D7   
3   A0A0C5Q4Y6                    Ferruginol synthase 1   
4   A0A0C5Q4Y6                    Ferruginol synthase 1   
5   A0A0C5Q4Y6                    Ferruginol synthase 1   
6   A0A0C5QRZ2                      Ferruginol synthase   
7   A0A0C5QRZ2                      Ferruginol synthase   
8   A0A0C5QRZ2                      Ferruginol synthase   
9   A0A1D6HSP4              Dimethylnonatriene synthase   
10  A0A1D6HSP4              Dimethylnonatriene synthase   
11  A0A2H4DGV8     Germacrene A acid 8-beta-hydroxylase   
12  A0A2H4DGV8     Germacrene A acid 8-beta-hydroxylase   
13  A0A2H4DGV8     Germacrene A acid 8-beta-hydroxylase   
14  A0A2Z5D854                     Xanthotoxol synthase   
15  A0A2Z5D854                     Xanthotoxol synthase   
16  A0A343URW6      Tabersonine 6,7-epoxidase isoform 1   
17  A0A517FNB9  Cholesterol 22-monohydroxylase CYP90B52   
18  A0A517FNC5  Cholesterol 16,22-dihydroxylase CYP90G4   
19  A0A517FNC6  Cholesterol 22-monohydroxylase CYP90B51   

                                             reaction rheaID  \
0   3-hydroxybenzyl alcohol + O2 + reduced [NADPH-...  62212   
1   3-methylphenol + O2 + reduced [NADPH--hemoprot...  62208   
2   an organic molecule + O2 + reduced [NADPH--hem...  17149   
3   abieta-8,11,13-triene + O2 + reduced [NADPH--h...  48080   
4   ferruginol + O2 + reduced [NADPH--hemoprotein ...  55428   
5   miltiradiene + 2 O2 + 2 reduced [NADPH--hemopr...  66796   
6   abieta-8,11,13-triene + O2 + reduced [NADPH--h...  48080   
7   ferruginol + O2 + reduced [NADPH--hemoprotein ...  55428   
8   miltiradiene + 2 O2 + 2 reduced [NADPH--hemopr...  66796   
9   (6E,10E)-geranyllinalool + O2 + reduced [NADPH...  13545   
10  (3S,6E)-nerolidol + O2 + reduced [NADPH--hemop...  55424   
11  germacra-1(10),4,11(13)-trien-12-oate + O2 + r...  57964   
12  germacra-1(10),4,11(13)-trien-12-oate + O2 + r...  57968   
13  germacra-1(10),4,11(13)-trien-12-oate + O2 + r...  58032   
14  O2 + psoralen + reduced [NADPH--hemoprotein re...  68548   
15  6-methoxycoumarin + O2 + reduced [NADPH--hemop...  68564   
16  (-)-tabersonine + O2 + reduced [NADPH--hemopro...  61056   
17  cholesterol + O2 + reduced [NADPH--hemoprotein...  69839   
18  cholesterol + 2 O2 + 2 reduced [NADPH--hemopro...  72191   
19  cholesterol + O2 + reduced [NADPH--hemoprotein...  69839   

                               substrates Substrate CHEBI ID  \
0                 3-hydroxybenzyl alcohol        CHEBI_17069   
1                          3-methylphenol        CHEBI_17231   
2                     an organic molecule       CHEBI_142491   
3                   abieta-8,11,13-triene        CHEBI_86062   
4                              ferruginol        CHEBI_78274   
5                            miltiradiene        CHEBI_65037   
6                   abieta-8,11,13-triene        CHEBI_86062   
7                              ferruginol        CHEBI_78274   
8                            miltiradiene        CHEBI_65037   
9                (6E,10E)-geranyllinalool        CHEBI_74299   
10                      (3S,6E)-nerolidol        CHEBI_59958   
11  germacra-1(10),4,11(13)-trien-12-oate        CHEBI_61301   
12  germacra-1(10),4,11(13)-trien-12-oate        CHEBI_61301   
13  germacra-1(10),4,11(13)-trien-12-oate        CHEBI_61301   
14                               psoralen        CHEBI_27616   
15                      6-methoxycoumarin       CHEBI_178005   
16                        (-)-tabersonine        CHEBI_57893   
17                            cholesterol        CHEBI_16113   
18                            cholesterol        CHEBI_16113   
19                            cholesterol        CHEBI_16113   

                                     Substrate SMILES  \
0                   

In [33]:
merged_df_with_seqs['length'] = merged_df_with_seqs['sequence'].apply(len)
merged_df_with_seqs.head(20)

protein                                     name  \
0   A0A075TMP8       Cytochrome P450 monooxygenase patI   
1   A0A075TRL5       Cytochrome P450 monooxygenase patH   
2   A0A087X1C5             Putative cytochrome P450 2D7   
3   A0A0C5Q4Y6                    Ferruginol synthase 1   
4   A0A0C5Q4Y6                    Ferruginol synthase 1   
5   A0A0C5Q4Y6                    Ferruginol synthase 1   
6   A0A0C5QRZ2                      Ferruginol synthase   
7   A0A0C5QRZ2                      Ferruginol synthase   
8   A0A0C5QRZ2                      Ferruginol synthase   
9   A0A1D6HSP4              Dimethylnonatriene synthase   
10  A0A1D6HSP4              Dimethylnonatriene synthase   
11  A0A2H4DGV8     Germacrene A acid 8-beta-hydroxylase   
12  A0A2H4DGV8     Germacrene A acid 8-beta-hydroxylase   
13  A0A2H4DGV8     Germacrene A acid 8-beta-hydroxylase   
14  A0A2Z5D854                     Xanthotoxol synthase   
15  A0A2Z5D854                     Xanthotoxol synthase   
16  A0A343URW6      Tabersonine 6,7-epoxidase isoform 1   
17  A0A517FNB9  Cholesterol 22-monohydroxylase CYP90B52   
18  A0A517FNC5  Cholesterol 16,22-dihydroxylase CYP90G4   
19  A0A517FNC6  Cholesterol 22-monohydroxylase CYP90B51   

                                             reaction rheaID  \
0   3-hydroxybenzyl alcohol + O2 + reduced [NADPH-...  62212   
1   3-methylphenol + O2 + reduced [NADPH--hemoprot...  62208   
2   an organic molecule + O2 + reduced [NADPH--hem...  17149   
3   abieta-8,11,13-triene + O2 + reduced [NADPH--h...  48080   
4   ferruginol + O2 + reduced [NADPH--hemoprotein ...  55428   
5   miltiradiene + 2 O2 + 2 reduced [NADPH--hemopr...  66796   
6   abieta-8,11,13-triene + O2 + reduced [NADPH--h...  48080   
7   ferruginol + O2 + reduced [NADPH--hemoprotein ...  55428   
8   miltiradiene + 2 O2 + 2 reduced [NADPH--hemopr...  66796   
9   (6E,10E)-geranyllinalool + O2 + reduced [NADPH...  13545   
10  (3S,6E)-nerolidol + O2 + reduced [NADPH--hemop...  55424   
11  germacra-1(10),4,11(13)-trien-12-oate + O2 + r...  57964   
12  germacra-1(10),4,11(13)-trien-12-oate + O2 + r...  57968   
13  germacra-1(10),4,11(13)-trien-12-oate + O2 + r...  58032   
14  O2 + psoralen + reduced [NADPH--hemoprotein re...  68548   
15  6-methoxycoumarin + O2 + reduced [NADPH--hemop...  68564   
16  (-)-tabersonine + O2 + reduced [NADPH--hemopro...  61056   
17  cholesterol + O2 + reduced [NADPH--hemoprotein...  69839   
18  cholesterol + 2 O2 + 2 reduced [NADPH--hemopro...  72191   
19  cholesterol + O2 + reduced [NADPH--hemoprotein...  69839   

                               substrates Substrate CHEBI ID  \
0                 3-hydroxybenzyl alcohol        CHEBI_17069   
1                          3-methylphenol        CHEBI_17231   
2                     an organic molecule       CHEBI_142491   
3                   abieta-8,11,13-triene        CHEBI_86062   
4                              ferruginol        CHEBI_78274   
5                            miltiradiene        CHEBI_65037   
6                   abieta-8,11,13-triene        CHEBI_86062   
7                              ferruginol        CHEBI_78274   
8                            miltiradiene        CHEBI_65037   
9                (6E,10E)-geranyllinalool        CHEBI_74299   
10                      (3S,6E)-nerolidol        CHEBI_59958   
11  germacra-1(10),4,11(13)-trien-12-oate        CHEBI_61301   
12  germacra-1(10),4,11(13)-trien-12-oate        CHEBI_61301   
13  germacra-1(10),4,11(13)-trien-12-oate        CHEBI_61301   
14                               psoralen        CHEBI_27616   
15                      6-methoxycoumarin       CHEBI_178005   
16                        (-)-tabersonine        CHEBI_57893   
17                            cholesterol        CHEBI_16113   
18                            cholesterol        CHEBI_16113   
19                            cholesterol        CHEBI_16113   

                                     Substrate SMILES  \
0                   

In [34]:
#getting rid of 'CHEBI_:' in my df to simplify the foundation of CHEBI IDs and SMILES in reaction_data.csv file
merged_df_with_seqs['Substrate CHEBI ID'] = merged_df_with_seqs['Substrate CHEBI ID'].str.replace("CHEBI_","")
merged_df_with_seqs['Product CHEBI ID'] = merged_df_with_seqs['Product CHEBI ID'].str.replace("CHEBI_","")



In [35]:
merged_df_with_seqs.drop(columns='participants', inplace=True)
merged_df_with_seqs.to_csv('reactionsfromjson.csv', index=False)